test/integration/targets/nxos_telemetry/tests/common/deleted.yaml

```diff
@@ -0,0 +1,109 @@
+        - { id: 2, data_source: NX-API, path: {name: sys/bgp/inst, depth: unbounded, query_condition: foo, filter_condition: foo}}
+        - { id: 55, data_source: DME, path: {name: 'sys/bgp/inst/dom-default/peer-[10.10.10.11]/ent-[10.10.10.11]', depth: 0, query_condition: foo, filter_condition: foo}}
+        - { id: 55, data_source: DME, path: {name: sys/ospf, depth: 0, query_condition: foo, filter_condition: 'or(eq(ethpmPhysIf.operSt,"down"),eq(ethpmPhysIf.operSt,"up"))'}}
+      subscriptions:
+        - { id: 44, destination_group: 10, sensor_group: {id: 8, sample_interval: 2000}}
+        - { id: 44, destination_group: 2, sensor_group: {id: 2, sample_interval: 2000}}
+        - { id: 55, destination_group: 10, sensor_group: {id: 55, sample_interval: 2000}}
+    provider: "{{ connection }}"
+
+- block:
+  - name: Gather Telemetry Facts Before Changes
+    nxos_facts: &facts
+      gather_subset:
+        - '!all'
+        - '!min'
+      gather_network_resources:
+        - telemetry
+
+  - name: Telemetry subscription - deleted
+    nxos_telemetry: &deleted
+      state: 'deleted'
+      config:
+        certificate:
+          key: /bootflash/server.key
+          hostname: localhost
+        compression: gzip
+        source_interface: Loopback55
+        vrf: management
+        destination_groups:
+          - id: 2
+            destination:
+              ip: 192.168.0.1
+              port: 50001
+              protocol: grpc
+              encoding: gpb
+          - { id: 2, destination: {ip: 192.168.0.2, port: 60001, protocol: gRPC, encoding: GPB}}
+          - { id: 10, destination: {ip: 192.168.0.1, port: 50001, protocol: Grpc, encoding: gPB}}
+          - { id: 10, destination: {ip: 192.168.0.2, port: 60001, protocol: gRPC, encoding: gpb}}
+        sensor_groups:
+          - { id: 8, data_source: NX-API, path: {name: sys/bgp, depth: 0, query_condition: foo, filter_condition: foo}}
+          - { id: 2, data_source: NX-API, path: {name: sys/bgp/inst, depth: unbounded, query_condition: foo, filter_condition: foo}}
+          - { id: 55, data_source: DME, path: {name: 'sys/bgp/inst/dom-default/peer-[10.10.10.11]/ent-[10.10.10.11]', depth: 0, query_condition: foo, filter_condition: foo}}
+          - { id: 55, data_source: DME, path: {name: sys/ospf, depth: 0, query_condition: foo, filter_condition: 'or(eq(ethpmPhysIf.operSt,"down"),eq(ethpmPhysIf.operSt,"up"))'}}
+        subscriptions:
+          - { id: 44, destination_group: 10, sensor_group: {id: 8, sample_interval: 2000}}
+          - { id: 44, destination_group: 2, sensor_group: {id: 2, sample_interval: 2000}}
+          - { id: 55, destination_group: 10, sensor_group: {id: 55, sample_interval: 2000}}
+      provider: "{{ connection }}"
+    register: result
+
+  - assert:
+      that:
+        - "(ansible_facts.network_resources.telemetry|dict2items)|symmetric_difference(result.before|dict2items)|length == 0"
```

Comment:
confused... you got a `before` snapshot of the config in the setup play, then `deleted` the config above.
Shouldn't the telemetry config now be removed?